VZ-10825: Add opensearchOperator as a high level component

saketmht opened the pull request:
List of changes:

<h2>CRD Changes</h2>

- Added opensearchOperator as a high level component
- It will now managed the existing opensearch and opensearchDashboards component
- opensearchOperator component doesn't have any overrides

<h2>OpenSearchOperator Install</h2><h3>PreInstall</h3><ol><li>Create the <code>verrazzano-logging</code> namespace</li><li>Handle legacyOS<ol><li>Set existing PVs to retain if they exist</li><li>Disable exiting OS and OSD via VMI</li><li>Delete the master nodes PVCs as they are created by sts so not deleted by default</li><li>Create new PVCs and set them to claim the existing PVs</li><li>Check if the new PVCs are bound to the existing PVs</li></ol></li></ol><p>Note:- We are not maintaining any state for the upgrade/pv-migration. The functions were designed to be re-entrant so that they can be run again in-case of errors and continue from where it left. These functions will be invoked during fresh installs as well but will have no effect as there will be no exiting PVs</p><h3>Install</h3><ol><li>Install the opensearch-operator helm chart in <code>verrazzano-logging</code> namespace</li><li>This will create the opensearch cluster as well</li></ol><h3>PostInstall</h3><ol><li>Set the reclaim policy of the PVs to what it was before</li></ol><p><br /></p><h2>
<h2>AuthProxy Changes</h2><p>Since the OpenSearch service will be changing with the new OpenSearch, the backed service in the authproxy needs to changed as well. The new services are passed thorough value overrides in the helm. These values are newly added.</p>

```
  opensearch:
    protocol: https
    namespace: verrazzano-logging
    service: opensearch
    osdService: opensearch-dashboards 
```

pfmackin commented on cluster-operator/controllers/vmc/jaeger.go:
```diff
@@ -70,21 +70,10 @@ func (r *VerrazzanoManagedClusterReconciler) getJaegerOpenSearchConfig(vzList *v
-			return jc, err
-		}
-		// Get the CA bundle needed to connect to the admin keycloak
-		jc.CA, err = r.getAdminCaBundle()
```
why did you remove this?

saketmht replied:
Its not removed. I actually deleted the else if block below. But since they had almost identical code, it is showing this part as deleted.

pfmackin commented on platform-operator/controllers/verrazzano/component/common/namespace.go:
```diff
@@ -140,3 +141,11 @@ func CheckExistingNamespace(ns []corev1.Namespace, includeNamespace func(*corev1
+func DoesNamespaceExists(client clipkg.Client, name string) bool {
+	ns := corev1.Namespace{}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: name}, &ns); err != nil {
+		return false
```
need to check for isNotFound error

saketmht replied:
Even if it is not found, this would have to return false

fredtibbitts replied:
Per Paul's comment, this function should return `(bool, error)` and return an error if there's an actual error fetching the namespace (ignoring not found).

michael-cico commented on platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:
```diff
@@ -466,6 +470,13 @@ type ComponentSpec struct {
+type OpenSearchOperatorComponent struct {
+	// If true, then OpenSearchOperator will be installed
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
```
If this is a Helm chart, it should support `InstallOverrides` right?

saketmht replied:
We decided against it. As per discussions with Mike/Kevin, we might want to move towards a more restrictive API in the future (VerrazzanoLogging CR). So if we support overrides now, it might be more difficult to move towards that later.

adiforluls replied:
Are we also going to prohibit the use of overrides for the "operator" as well?

adiforluls replied:
I think for configs like pod/container security, will have to externalize those things in the CRD since we're not going to make helm values configurable.

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/legacyOS.go:
```diff
@@ -0,0 +1,128 @@
+	vmiList := vmov1.VerrazzanoMonitoringInstanceList{}
+	err := ctx.Client().List(context.TODO(), &vmiList)
+
+	if _, ok := err.(*meta.NoKindMatchError); ok {
```
suggest to use `meta.IsNoMatchError`

saketmht replied:
Updated

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+}
+
+func getMasterNode(ctx spi.ComponentContext) string {
+	for _, node := range ctx.EffectiveCR().Spec.Components.Elasticsearch.Nodes {
```
Is it necessary to consider the edge case where there are multiple master node groups in this function?

saketmht replied:
Any one master node will do

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+	if IsUpgrade(ctx) || IsSingleMasterNodeCluster(ctx) {
+		kvs = append(kvs, bom.KeyValue{
+			Key:   `opensearchCluster.bootstrap.additionalConfig.cluster\.initial_master_nodes`,
+			Value: fmt.Sprintf("%s-%s-0", clusterName, getMasterNode(ctx)),
```
What if this returns an empty string?

saketmht replied:
That would be an invalid configuration. I will add a check and return error if no master nodes are present

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+			ingressAnnotations[`external-dns\.alpha\.kubernetes\.io/ttl`] = "60"
+		}
+
+		kvs, _ = appendOSIngressOverrides(ingressAnnotations, dnsSubDomain, ingressClassName, kvs)
```
why are the errors being ignored?

saketmht replied:
It was not really returning any error since we I was just appending values to kvs

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+}
+
+// appendOSDIngressOverrides appends the additional overrides for OpenSearchDashboards ingress
+func appendOSDIngressOverrides(ingressAnnotations map[string]string, dnsSubDomain, ingressClassName string, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
```
this function doesn't appear to return an error

saketmht replied:
Updated

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+}
+
+// appendOSIngressOverrides appends the additional overrides for OpenSearch ingress
+func appendOSIngressOverrides(ingressAnnotations map[string]string, dnsSubDomain, ingressClassName string, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
```
This function also doesn't appear to return an error. It's also very similar to the prior function, are you able to make it more generic?

saketmht replied:
Updated

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+	return ready.DeploymentsAreReady(ctx.Log(), ctx.Client(), deployments, 1, getPrefix(ctx))
+}
+
+// IsSingleMasterNodeCluster returns true if the cluster has a single mater node
```
typo in comment

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+			}
+		}
+	}
+	return replicas <= 1
```
Is it a single master cluster if there are zero master nodes? This is some invalid topology which may come from user error

saketmht replied:
If there are no master nodes, it is an invalid configuration but I don't think it makes sense to check here

anders-swanson commented on platform-operator/controllers/verrazzano/component/opensearchoperator/pvc.go:
```diff
@@ -0,0 +1,269 @@
+)
+
+// getOSPersistentVolumes returns the list of older PersistentVolumes used by VMO OpenSearch
+func getOSPersistentVolumes(ctx spi.ComponentContext, nodes []vzapi.OpenSearchNode) ([]v1.PersistentVolume, error) {
```
Can you use the VMI to get the PVCs?

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearch/opensearch.go:
```diff
@@ -87,13 +94,19 @@ func isOSNodeReady(ctx spi.ComponentContext, node vzapi.OpenSearchNode, prefix s
 
 	// If a node has the master role, it is a statefulset
-	if hasRole(node.Roles, vmov1.MasterRole) {
+	// If the operator is managing OS, then all nodes are statefulset
```
nit, might want to distinguish between opster-managed and the VMO in these comments.  They're both operators.

michael-cico commented on pkg/vzcr/enabled.go:
```diff
@@ -697,3 +697,17 @@ func IsAlertmanagerEnabled(cr runtime.Object, ctx spi.ComponentContext) (bool, e
 	return false, nil
 }
+
+// IsOpenSearchOperatorEnabled returns false only if OpenSearchOperator is explicitly disabled in the CR
```
Do we need to change the `IsVMOEnabled` func to not check for Opensearch?  Or do we need to keep having it return true if Opensearch is enabled so that it continues managing things until the upgrade/swap is done with Opster?

saketmht replied:
I am testing a few scenarios but for now I think we can leave it as it is.

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearchdashboards/opensearchdashboards.go:
```diff
@@ -15,11 +16,20 @@ import (
 		{
-			Name:      kibanaDeployment,
-			Namespace: ComponentNamespace,
+			Name:      "opensearch-dashboards",
```
nit, should define a constant for this.

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator_component.go:
```diff
@@ -0,0 +1,134 @@
+			SupportsOperatorInstall:   true,
+			SupportsOperatorUninstall: true,
+			Dependencies:              []string{networkpolicies.ComponentName, certmanager.ComponentName, nginx.ComponentName},
+			AppendOverridesFunc:       AppendOverrides,
```
`AppendOverrides` func probably doesn't need to be exported

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearchoperator/pvc.go:
```diff
@@ -0,0 +1,269 @@
+		}
+		// If old reclaim policy is not already set, set it
+		// So that it can be changed back to its original value later
+		_, ok := pv.Labels[constants.OldReclaimPolicyLabel]
```
if you used `controllerruntime.CreateOrUpdate()` you could just set values and that func will only update the CR if anything's changed.

adiforluls replied:
+1

saketmht replied:
The thing here is that these functions can be called multiple times. So oldReclaimPolicy should only be set when it PV is set to RETAIN the first time. SInce the next time, its policy would be retain(instead of delete) and we don't want to set the oldReclaimPolicy to RETAIN as well.

michael-cico commented on platform-operator/internal/config/config.go:
```diff
@@ -182,6 +183,13 @@ func GetHelmClusterOpChartsDir() string {
 	return filepath.Join(instance.VerrazzanoRootDir, helmClusterOpChartsDirSuffix)
 }
 
+func GetHelmOpenSearchOpChartsDir() string {
```
Didn't know we had all these component-chart helpers... but should we use this to set the chart location when the OS operator component is instantiated?

saketmht replied:
I am not sure what you mean by this. Every component has it this way only.

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+
+// AppendOverrides appends the additional overrides for install
+func AppendOverrides(ctx spi.ComponentContext, _ string, _ string, _ string, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
+	// TODO: Image overrides once the BFS images are done
```
aren't the image overrides handled by the `HelmComponent` code?  Or are there other images that can't be handled by that?

saketmht replied:
The image for opensearch-operator will be handled by HelmComponent itself. But since we are managing OS and OSD via this component now as well, we would need to append images for OS and OSD now here

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,498 @@
+
+	if osd != nil {
+		osdPlugins := osd.Plugins
+		if osdPlugins.Enabled && len(osdPlugins.InstallList) > 0 {
```
might be good to wrap this in an `IsXXXXEnabled` func internally to handle the nil-pointer case.

saketmht replied:
Plugins.Enabled is not a pointer actually

adiforluls commented on platform-operator/controllers/verrazzano/component/common/vmi.go:
```diff
@@ -78,6 +78,36 @@ func DeleteVMI(ctx spi.ComponentContext) error {
+			return true
+		}
+	}
+	return false
```
If the error is `IsNotFound`, I think this should return false, but what about other type of errors? Would you want to retry in that case?
Maybe the func should return `(bool, error)`, that way the caller can inspect the error value also along with bool value.

saketmht replied:
This is called from the Ready checks which don't return an error. But I have added an error log

adiforluls commented on platform-operator/controllers/verrazzano/component/common/vmi.go:
```diff
@@ -78,6 +78,36 @@ func DeleteVMI(ctx spi.ComponentContext) error {
+			return true
+		}
+	}
+	return false
```
Same question/comment as above.

fredtibbitts replied:
Agree with @adiforluls, these functions should not eat errors. Return the error and the caller can decide to ignore or not.

saketmht replied:
Updated

saketmht commented on platform-operator/controllers/verrazzano/component/common/vmi.go:
```diff
@@ -78,6 +78,36 @@ func DeleteVMI(ctx spi.ComponentContext) error {
+	if vzcr.IsOpenSearchEnabled(ctx.EffectiveCR()) {
+		systemVMI := vmov1.VerrazzanoMonitoringInstance{}
+		err := ctx.Client().Get(context.TODO(), types.NamespacedName{Name: VMIName, Namespace: globalconst.VerrazzanoSystemNamespace}, &systemVMI)
+		if err != nil && errors.IsNotFound(err) {
```
Probably just check for newer opensearch sts in logging ns since that is the default going forward

saketmht commented:
I have refactored the code based on comments and to add overrides for the opensearchOperator component. For this, I removed the cluster chart from the operator helm and added a yaml manifest file for the cluster to which we pass some dynamic values and apply that to create the OS cluster

michael-cico commented on platform-operator/controllers/verrazzano/component/common/vmi.go:
```diff
@@ -78,6 +79,42 @@ func DeleteVMI(ctx spi.ComponentContext) error {
+			if meta.IsNoMatchError(err) || errors.IsNotFound(err) {
+				return false
+			}
+			ctx.Log().Errorf("Failed to get VMI, considering legacy OS to be disabled")
```
You should use `ErrorfThrottled` so the message is throttled.

michael-cico replied:
Can be done in a second pass if you prefer.

michael-cico commented on platform-operator/controllers/verrazzano/component/common/vmi.go:
```diff
@@ -78,6 +79,42 @@ func DeleteVMI(ctx spi.ComponentContext) error {
+			if meta.IsNoMatchError(err) || errors.IsNotFound(err) {
+				return false
+			}
+			ctx.Log().Errorf("Failed to get VMI, considering legacy OSD to be disabled")
```
Use `ErrorfThrottled`

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator_component.go:
```diff
@@ -114,6 +117,31 @@ func (o opensearchOperatorComponent) PreInstall(ctx spi.ComponentContext) error
+		err = yamlApplier.ApplyFT(filePath, args)
+
+		if err != nil {
+			return ctx.Log().ErrorfNewErr("Failed to substitute template values for OpenSearchCluster CR: %v", err)
```
Use `ErrorfThrottledNewErr`

anders-swanson commented on pkg/k8sutil/apply_yaml.go:
```diff
@@ -478,3 +479,15 @@ func nindent(indent int, s string) string {
+
+func multiLineIndent(indentNum int, aff string) string {
+	var b = make([]byte, indentNum)
+	for i := 0; i < indentNum; i++ {
```
see [strings.Repeat](https://pkg.go.dev/strings#Repeat) for a more idiomatic way of creating strings like this. It also is not left up to the reader to know that `32` here is used as the ascii code for a space

anders-swanson replied:
i still don't get it, we use nindent defined above for multi line indents, the input is split on newline/linefeed characters

saketmht replied:
nindent doesn't work for multi-line inputs. I tried it and didn't get the desired result so used this instead. If I want to format a yaml at 4 space and I use nindent, the result will be
```
    - component: es-master         // First line indented
replicas: 3
roles:
- ...
- component: ...                  // Rest of line not indented
```
I didn't really write. I just copied it over from https://github.com/verrazzano/verrazzano/blob/2ae22d8487738ad9840b4742f25c09f93c102a00/platform-operator/apis/verrazzano/v1alpha1/convert_istio_to.go#L212 which is something you wrote actually. 
Your comment
> / Define the IstioOperator template which is used to insert the generated YAML values.
//
// NOTE: The go template rendering doesn't properly indent the multi-line YAML value
// For example, the template fragment only indents the fist line of values

fredtibbitts commented on platform-operator/controllers/verrazzano/component/common/vmi.go:
```diff
@@ -6,6 +6,7 @@ package common
 import (
 	"context"
 	"fmt"
+	"k8s.io/apimachinery/pkg/api/meta"
```
Imports not formatted correctly.

fredtibbitts commented on platform-operator/controllers/verrazzano/component/common/vmi.go:
```diff
@@ -78,6 +79,42 @@ func DeleteVMI(ctx spi.ComponentContext) error {
 }
 
+// IsLegacyOS returns true if the OS that is running is managed by VMO
+func IsLegacyOS(ctx spi.ComponentContext) bool {
```
Return `(bool, error)`, and let the caller decide whether it wants to ignore errors.

fredtibbitts commented on platform-operator/controllers/verrazzano/component/opensearch/opensearch.go:
```diff
@@ -5,6 +5,7 @@ package opensearch
 
 import (
 	"fmt"
+	"github.com/verrazzano/verrazzano/platform-operator/constants"
```
Import formatting.

fredtibbitts commented on platform-operator/controllers/verrazzano/component/opensearch/opensearch_component.go:
```diff
@@ -5,6 +5,7 @@ package opensearch
 
 import (
 	"fmt"
+	"github.com/verrazzano/verrazzano/platform-operator/controllers/verrazzano/component/opensearchoperator"
```
Import formatting.

fredtibbitts commented on platform-operator/controllers/verrazzano/component/opensearch/opensearch_component.go:
```diff
@@ -103,24 +103,12 @@ func NewComponent() spi.Component {
 }
 
 // Install OpenSearch component install processing
 func (o opensearchComponent) Install(ctx spi.ComponentContext) error {
```
I think you can just delete these functions since they just return `nil`.

saketmht replied:
Since OS and OSD component do not implement Helm component, these need to have some implementation defined

fredtibbitts replied:
Oh, right. I didn't look to see that it wasn't a Helm component. Thanks.

fredtibbitts commented on platform-operator/controllers/verrazzano/component/opensearch/opensearch_test.go:
```diff
@@ -5,6 +5,7 @@ package opensearch
 
 import (
 	"fmt"
+	globalconst "github.com/verrazzano/verrazzano/pkg/constants"
```
Import formatting.

fredtibbitts commented on platform-operator/controllers/verrazzano/component/opensearchdashboards/opensearchdashboards.go:
```diff
@@ -1,10 +1,11 @@
 
 import (
 	"fmt"
+	"github.com/verrazzano/verrazzano/platform-operator/constants"
```
Import formatting.

fredtibbitts commented on platform-operator/controllers/verrazzano/component/opensearchdashboards/opensearchdashboards_component.go:
```diff
@@ -105,24 +107,12 @@ func NewComponent() spi.Component {
 }
 
 // Install OpenSearch-Dashboards component install processing
 func (d opensearchDashboardsComponent) Install(ctx spi.ComponentContext) error {
```
Delete these functions since they just return `nil`.

saketmht replied:
This doesn't extend Helm Component so these need to have implementations defined

fredtibbitts commented on platform-operator/controllers/verrazzano/component/opensearchdashboards/opensearchdashboards_component_test.go:
```diff
@@ -4,6 +4,8 @@ package opensearchdashboards
 
 import (
 	"context"
+	vmov1 "github.com/verrazzano/verrazzano-monitoring-operator/pkg/apis/vmcontroller/v1"
```
Import formatting.

fredtibbitts commented:
Please review the code smells. At least a few of those should be addressed.

michael-cico commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator_component.go:
```diff
@@ -0,0 +1,173 @@
+			IgnoreNamespaceOverride:   true,
+			SupportsOperatorInstall:   true,
+			SupportsOperatorUninstall: true,
+			Dependencies:              []string{networkpolicies.ComponentName, certmanager.ComponentName, nginx.ComponentName},
```
You may want to add a dependecy on the `clusterIssuer` component instead of `certManager`.  the `clusterIssuer` configuration controls the configuration of the `clusterResouceNamespace` we install certManager with.

michael-cico replied:
Or, just add `clusterIssuer` instead of replacing `certManager`

pfmackin commented on platform-operator/controllers/obsolete/reconcile/controller.go:
```diff
@@ -911,7 +911,36 @@ func (r *Reconciler) watchResources(namespace string, name string, log vzlog.Ver
 
 	log.Debugf("Watching the Rancher global data namespace to create admin cloud credential")
-	return r.watchRancherGlobalDataNamespace(namespace, name)
+	if err := r.watchRancherGlobalDataNamespace(namespace, name); err != nil {
```
This code is obsolete and not executeed. Any changes you make here have no effect.

pfmackin commented on platform-operator/controllers/verrazzano/component/common/opensearch_operator.go:
```diff
@@ -0,0 +1,567 @@
+	return replicas <= 1
+}
+
+// IsUpgrade returns true if we are upgrading from <=1.6.x to 2.x
```
comment is wrong, shoudld <= 1.6.x to > 1.6.x

pfmackin commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,260 @@
+// Like OpenSearchRoles, OpenSearchUserRolesBindings
+// Since the operator adds a finalizer to these resources, they need to deleted before the operator is uninstalled
+func (o opensearchOperatorComponent) deleteRelatedResource() error {
+	client, err := k8sutil.GetDynamicClient()
```
Change this to do it the same way that cert-manager does it so that the finalizer is removed.  See https://github.com/verrazzano/verrazzano/blob/master/platform-operator/controllers/verrazzano/component/certmanager/certmanager/certmanager_component.go#L186

pfmackin commented on platform-operator/controllers/verrazzano/component/opensearchoperator/opensearchoperator.go:
```diff
@@ -0,0 +1,260 @@
+// buildIngressOverrides builds the overrides required for the OpenSearch and OpenSearchDashboards ingresses
+func buildIngressOverrides(ctx spi.ComponentContext, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
+	if vzcr.IsNGINXEnabled(ctx.EffectiveCR()) {
+
```
remove line 149

pfmackin commented on platform-operator/helm_config/overrides/opensearch-operator-values.yaml:
```diff
@@ -0,0 +1,60 @@
+    runAsNonRoot: true
+    runAsUser: 65532
+  image:
+    repository: public.ecr.aws/opsterio/opensearch-operator
```
why is in the overrides.  We get this image from the BOM don't we?

saketmht replied:
removed

pfmackin commented on tests/e2e/security/netpol/network_policy_test.go:
```diff
@@ -478,11 +482,11 @@ var _ = t.Describe("Test Network Policies", Label("f:security.netpol"), func() {
-				err := testAccess(metav1.LabelSelector{MatchLabels: map[string]string{"app": "netpol-test"}}, "netpol-test", metav1.LabelSelector{MatchLabels: map[string]string{"app": "system-es-master"}}, vzconst.VerrazzanoSystemNamespace, 9200, false, false)
-				Expect(err).To(BeNil(), fmt.Sprintf("FAIL: Negative test vmi-system-es-master ingress rules failed: reason = %s", err))
-			},
+			//func() {
```
you should do a follow up JIRA to remove these commented out tests to get them to work.  If for some reason they are no longer valid then remove them.

saketmht replied:
Sure

pfmackin commented on tests/e2e/pkg/elasticsearch.go:
```diff
@@ -344,37 +343,22 @@ func GetSystemOpenSearchIngressURL(kubeconfigPath string) string {
 		return ""
 	}
-	// Return the os ingress host from the verrazzano-logging namespace if it exists
+	// Return the newer OS ingress if >= 2.0.0
```
2.0.0 is wrong.

pfmackin commented on tests/e2e/pkg/elasticsearch.go:
```diff
@@ -344,37 +343,22 @@ func GetSystemOpenSearchIngressURL(kubeconfigPath string) string {
+	// Return the newer OS ingress if >= 2.0.0
 	// Else return the vmi ingress as usual
-	useloggingNSIngress, _ := osIngressInLoggingNSExists()
+	useNewOSIngress, _ := IsVerrazzanoMinVersion("2.0.0", kubeconfigPath)
```
2.0.0 is wrong.

pfmackin commented on tests/e2e/pkg/opensearch_dashboards.go:
```diff
@@ -133,14 +133,15 @@ func PostOpensearchDashboards(path string, body string, additionalHeaders ...str
 func getOpenSearchDashboardsURL(kubeconfigPath string) string {
 	clientset, err := GetKubernetesClientsetForCluster(kubeconfigPath)
 	isMinversion150, _ := IsVerrazzanoMinVersion("1.5.0", kubeconfigPath)
+	isMinversion200, _ := IsVerrazzanoMinVersion("2.0.0", kubeconfigPath)
```
2.0.0 is wrong.

pfmackin commented on tests/e2e/uninstall/verifycrds/verify_crds_suite_test.go:
```diff
@@ -10,12 +10,10 @@ import (
 func init() {
-	flag.BoolVar(&mySQLOperatorEnabled, "mySQLOperatorEnabled", true, "mySQLOperatorEnabled indicates whether the mySQLOperator component is enabled")
-	flag.BoolVar(&vmoEnabled, "vmoEnabled", true, "vmoEnabled indicates whether the Verrazzano monitoring operator is enabled")
+	flag.BoolVar(&isManagedCluster, "isManagedCluster", false, "isManagedCluster indicates if it is a managed-cluster profile or not")
```
we should test based on a profile.  We should check the CR for disabled/enabled components

saketmht replied:
This parameter is passed through the jenkins file where we are checking if it is a managed cluster or not

pfmackin commented on tests/e2e/uninstall/verifycrds/verify_crds_test.go:
```diff
@@ -114,7 +123,7 @@ var _ = t.Describe("Verify CRDs after uninstall.", Label("f:platform-lcm.unnstal
 		Fail(err.Error())
 	}
-	if vmoEnabled {
+	if !isManagedCluster {
```
we should test based on a profile.  We should check the CR for disabled/enabled components

pfmackin commented on tests/e2e/uninstall/verifycrds/verify_crds_test.go:
```diff
@@ -151,11 +160,14 @@ var _ = t.Describe("Verify CRDs after uninstall.", Label("f:platform-lcm.unnstal
 	})
 
-	if mySQLOperatorEnabled {
+	if !isManagedCluster {
```
we should test based on a profile.  We should check the CR for disabled/enabled components

pfmackin commented on tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go:
```diff
@@ -296,12 +328,44 @@ var _ = t.Describe("Checking if Verrazzano system components are ready, post-upg
+			func(namespace string, componentName string, stsName string) {
+				// Currently we have no way of determining if some components are installed by looking at the status (grafana)
+				// Because of this, make this test non-managed cluster only.
+				if vzcr.Spec.Profile == vzapi.ManagedCluster {
```
we should test based on a profile.  We should check the CR for disabled/enabled components

saketmht replied:
All the other tests are written in this fashion. Changing one would likely mean to change all the other existing tests as well.

pfmackin commented on tests/e2e/verify-infra/vmi/vmi_test.go:
```diff
@@ -154,11 +155,13 @@ var beforeSuite = t.BeforeSuiteFunc(func() {
 	}, waitTimeout, pollingInterval).ShouldNot(HaveOccurred())
 
-	elastic = vmi.GetOpensearch("system")
+	ok, _ := pkg.IsVerrazzanoMinVersion("2.0.0", kubeconfigPath)
```
2.0.0 is wrong

pfmackin commented on tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go:
```diff
@@ -188,6 +191,19 @@ var _ = t.Describe("Checking if Verrazzano system components are ready, post-upg
 						pkg.Log(pkg.Error, fmt.Sprintf("failed to find the verrazzano version: %v", err))
 						return false
 					}
+					isVersionAbove2_0_0, err := pkg.IsVerrazzanoMinVersion("2.0.0", kubeconfigPath)
```
Version 2.0 is wrong

saketmht replied:
Currently the version is 2.0 in master as well, will update once it is set to 1.7 in msater
